metrics/stackdriver_exporter.go

```diff
@@ -198,49 +198,60 @@ func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (v
 		nil
 }
 
-func sdCustomMetricsRecorder(mc metricsConfig) func(context.Context, []stats.Measurement, ...stats.Options) error {
+func sdCustomMetricsRecorder(mc metricsConfig, allowCustomMetrics bool) func(context.Context, []stats.Measurement, ...stats.Options) error {
 	gm := getMergedGCPMetadata(&mc)
 	metadataMap := map[string]string{
 		metricskey.LabelProject:     gm.project,
 		metricskey.LabelLocation:    gm.location,
 		metricskey.LabelClusterName: gm.cluster,
 	}
 	return func(ctx context.Context, mss []stats.Measurement, ros ...stats.Options) error {
-		// Filter the measuremenst array to only include permitted metrics.
 		i := 0
 		var templ *resourceTemplate
+		hasCustomMetrics := false
+
+		// This loop serves two purpose, filtering out custom metrics when allowCustomMetrics set to false,
+		// and find the template for the system metrics.
 		for _, m := range mss {
 			metricType := path.Join(mc.stackdriverMetricTypePrefix, m.Measure().Name())
-			if t, ok := metricToResourceLabels[metricType]; ok {
+			t, ok := metricToResourceLabels[metricType]
+			if ok || allowCustomMetrics {
 				mss[i] = m
 				i++
-				if templ != nil && templ != t {
-					return fmt.Errorf("mixed resource type measurements in one report: %v", mss)
+				hasCustomMetrics = hasCustomMetrics || !ok
+				if ok {
+					if templ != nil && templ != t {
+						return fmt.Errorf("mixed resource type measurements in one report: %v", mss)
+					}
+					templ = t
 				}
```

Comment:
After going over this with several truth tables, I think we want this (and then we don't need 235+

```golang

if (templ != nil || hasCustomMetrics) && templ != t {
  // Don't allow re-assignment of templ; if hasCustomMetrics is set, templ has been "assigned" to nil
  return fmt.Errorf("mixed resource type measurements in one report: %v", mss)
}
templ = t